src/main/java/mate/academy/onlinebookstore/service/book/BookServiceImpl.java

```diff
@@ -23,23 +28,37 @@ public class BookServiceImpl implements BookService {
 
     @Override
     public BookDto save(CreateBookRequestDto bookRequestDto) {
+        Optional<Book> existingBook = bookRepository.findByIsbnIncludingDeleted(
+                bookRequestDto.isbn());
+        if (existingBook.isPresent()) {
+            Book book = existingBook.get();
+            if (book.isDeleted()) {
+                throw new DataProcessingException("Can't save book. A book with this ISBN: "
+                        + bookRequestDto.isbn() + " already exists but is marked as deleted.");
+            }
+            throw new DataProcessingException("Can't save book. A book with this ISBN: "
+                    + bookRequestDto.isbn() + " already exists in the database.");
+        }
```

Comment:
does the user need these details? I think it's enough to know that the ISBN was wrong.